packages/http-client/src/remote-index-api.ts

```diff
@@ -1,7 +1,29 @@
 import type { BaseQuery, IndexApi, Page, Pagination, StreamState } from '@ceramicnetwork/common'
+import { StreamUtils, fetchJson } from '@ceramicnetwork/common'
 
+/**
+ * IndexAPI implementation on top of HTTP endpoint.
+ */
 export class RemoteIndexApi implements IndexApi {
-  queryIndex(query: BaseQuery & Pagination): Promise<Page<StreamState>> {
-    throw new Error(`NotImplemented: RemoteIndexApi::queryIndex`)
+  private readonly _collectionURL: URL
+
+  constructor(apiUrl: URL, private readonly fetchJsonFn: typeof fetchJson = fetchJson) {
+    this._collectionURL = new URL('./collection', apiUrl)
+  }
+
+  /**
+   * Issue a query to `/collection` endpoint.
+   */
+  async queryIndex(query: BaseQuery & Pagination): Promise<Page<StreamState>> {
```

Comment:
I just realized this function (and therefore also the signature definition in `IndexApi` and the implementation in `LocalIndexApi`) shouldn't be returning `StreamState` objects but actually should be returning Stream instances of the appropriate streamtype, same as how `CeramicApi.load()` transforms the resulting StreamStates into Stream instances

Reply:
Unless you want to make the IndexApi private in `Ceramic` and `CeramicClient` and have a `query()` method on `CeramicApi` that wraps `IndexApi.queryIndex` and transforms the StreamStates to stream instances.  But whatever the end-user application dev consumes in their application code should be returning Stream instances of the proper StreamType.

Reply:
> should be returning Stream instances of the proper StreamType

1. I really thought we agreed on returning StreamState here.
2. Returning actual Streams would be a huge waste of compute resources.

Reply:
@ukstv are you okay with going with what we discussed in discord?
`IndexApi.queryIndex` stays as-is, but `IndexApi` becomes private within the `CeramicApi` implementations.  Then we add a new `query` method to `CeramicApi` that is more user-friendly: returns an AsyncIterable, handles requesting subsequent batches internally, and returns stream instances one at a time

Reply:
decision from standup: this will basically stay as-is in this PR.  `IndexApi.queryIndex` is the primary way we expose querying functionality.  It is designed to be consumed by the GraphQL runtime tooling which will handle paging and translating StreamStates to Stream instances.

In the future we can add an API that wraps `IndexApi.queryIndex` into something more user-friendly, if there is demand for it